src/components/VisualizationLayer.js

```diff
@@ -9,70 +9,66 @@ import { hexToRgb } from "./svg/SvgHelper";
 import PropTypes from "prop-types";
 
 class VisualizationLayer extends React.PureComponent {
-  constructor(props) {
-    super(props);
+  static defaultProps = { position : [0, 0] };
 
-    this.canvasDrawing = [];
+  canvasDrawing = [ ];
 
-    this.state = {
-      canvasDrawing: [],
-      dataVersion: "",
-      renderedElements: []
-    };
-  }
+  state = {
+    canvasDrawing: [],
+    dataVersion: "",
+    renderedElements: []
+  };
 
   componentDidUpdate() {
-    const adjustedPosition = this.props.position || [0, 0];
-    let context;
-    if (this.props.canvasContext && !this.props.disableContext) {
-      context = this.props.canvasContext.getContext("2d");
+    if (this.props.disableContext || !this.props.canvasContext || !this.canvasDrawing.length)
```

Comment:
None of the below code can work without `canvasContext` and `context` so it makes since to collapse the if conditions into one. In fact, the bottom if condition in the previous commit would throw an exception in some cases since it could be run without `let context` being assigned to (I discovered that in writing the above test)